### PR TITLE
null safety widgets instance

### PR DESCRIPTION
**Pull request by nathanielajayi7**
The instance of WidgetsBinding cannot be accessed unconditionally

### CODE REVIEW

File: lib/src/flutter_typeahead.dart [hunk 1]
```diff
@@ -751,7 +751,7 @@ class _TypeAheadFieldState<T> extends State<TypeAheadField<T>>
   void dispose() {
     this._suggestionsBox!.close();
     this._suggestionsBox!.widgetMounted = false;
-    WidgetsBinding.instance.removeObserver(this);
+    WidgetsBinding.instance?.removeObserver(this);
```
Comment (zaikavova) [inline lib/src/flutter_typeahead.dart:754]: Flutter 3.0 now gives the required value, so this is not needed.

File: lib/src/cupertino_flutter_typeahead.dart [hunk 2]
```diff
@@ -602,7 +602,7 @@ class _CupertinoTypeAheadFieldState<T> extends State<CupertinoTypeAheadField<T>>
       }
     });
 
-    WidgetsBinding.instance.addPostFrameCallback((duration) {
+    WidgetsBinding.instance?.addPostFrameCallback((duration) {
```
Comment (zaikavova) [inline lib/src/cupertino_flutter_typeahead.dart:605]: Flutter 3.0 now gives the required value, so this is not needed.

File: lib/src/cupertino_flutter_typeahead.dart [hunk 3]
```diff
@@ -568,7 +568,7 @@ class _CupertinoTypeAheadFieldState<T> extends State<CupertinoTypeAheadField<T>>
   @override
   void initState() {
     super.initState();
-    WidgetsBinding.instance.addObserver(this);
+    WidgetsBinding.instance?.addObserver(this);
```
Comment (zaikavova) [inline lib/src/cupertino_flutter_typeahead.dart:571]: Flutter 3.0 now gives the required value, so this is not needed.

File: lib/src/cupertino_flutter_typeahead.dart [hunk 4]
```diff
@@ -555,7 +555,7 @@ class _CupertinoTypeAheadFieldState<T> extends State<CupertinoTypeAheadField<T>>
   void dispose() {
     this._suggestionsBox!.close();
     this._suggestionsBox!.widgetMounted = false;
-    WidgetsBinding.instance.removeObserver(this);
+    WidgetsBinding.instance?.removeObserver(this);
```
Comment (zaikavova) [inline lib/src/cupertino_flutter_typeahead.dart:558]: Flutter 3.0 now gives the required value, so this is not needed.

File: lib/src/flutter_typeahead.dart [hunk 5]
```diff
@@ -764,7 +764,7 @@ class _TypeAheadFieldState<T> extends State<TypeAheadField<T>>
   @override
   void initState() {
     super.initState();
-    WidgetsBinding.instance.addObserver(this);
+    WidgetsBinding.instance?.addObserver(this);
```
Comment (zaikavova) [inline lib/src/flutter_typeahead.dart:767]: Flutter 3.0 now gives the required value, so this is not needed.

File: lib/src/flutter_typeahead.dart [hunk 6]
```diff
@@ -800,7 +800,7 @@ class _TypeAheadFieldState<T> extends State<TypeAheadField<T>>
       }
     });
 
-    WidgetsBinding.instance.addPostFrameCallback((duration) {
+    WidgetsBinding.instance?.addPostFrameCallback((duration) {
```
Comment (zaikavova) [inline lib/src/flutter_typeahead.dart:803]: Flutter 3.0 now gives the required value, so this is not needed.

**Review by zaikavova — CHANGES_REQUESTED** [review]
Remove downgrading changes

**Comment by clragon** [comment]
this merge request would undo migration to flutter 3.0
please update your flutter version.
if you cannot, it is possible to introduce a backwards compatible version of this code,
as instructed here:

https://docs.flutter.dev/development/tools/sdk/release-notes/release-notes-3.0.0#your-code

**Comment by sjmcdowall** [comment]
As per clragon -- this PR would undo the 3.0 flutter  .. so won't be done ..

**Pull request closed without merging**